src/Eloquent/Model.php

```diff
@@ -284,23 +278,25 @@ public function setAttribute($key, $value)
     /** @inheritdoc */
     protected function asDecimal($value, $decimals)
     {
-        try {
-            $value = (string) BigDecimal::of((string) $value)->toScale((int) $decimals, RoundingMode::HALF_UP);
-
-            return new Decimal128($value);
```

Comment:
:+1: Good catch, the method `asDecimal` must return a string, as mandated in the phpdoc of the parent method.